R/initialize_modules.R

```diff
@@ -0,0 +1,664 @@
+#' Initializes a distribution module by setting up its fields based on the
+#' distribution name and type. Supports both "data" and "process" types.
+#' @param module_input A list. Contains parameters for initializing the distribution.
+#' @param distribution_name A character. Name of the distribution to initialize.
+#' @param distribution_type A character. Type of distribution, either "data" or "process".
+#' @param linked_ids A vector. Named vector of linked IDs required for the distribution,
+#' such as data_link and fleet_link for setting up index distribution.
+#' @rdname initialize_module
+#' @return The initialized distribution module as an object.
+#' @noRd
+initialize_distribution <- function(module_input, distribution_name,
+                                    distribution_type = c("data", "process"), linked_ids) {
+  # Check if distribution_name is provided
+  if (is.null(distribution_name)) {
+    return(NULL)
+  }
+
+  # Validate module_input
+  if (!is.list(module_input)) {
+    cli::cli_abort("{.var module_input} must be a list.")
+  }
+
+  # Validate distribution_type as "data" or "process"
+  valid_distribution_types <- c("data", "process")
+  check_valid_input(
+    input = distribution_type,
+    valid_options = valid_distribution_types,
+    arg_name = "distribution_type"
+  )
+
+  # Validate linked_ids as a named vector with required elements for "data" type
+  if (!is.vector(linked_ids) || !all(c("data_link", "fleet_link") %in% names(linked_ids))) {
+    cli::cli_abort("{.var linked_ids} must be a named vector containing 'data_link' and 'fleet_link' for 'data' distribution types.")
+  }
+
+  # Get distribution value and initialize the module
+  distribution_value <- get(distribution_name)
+  distribution_module <- new(distribution_value)
+  distribution_fields <- names(distribution_value@fields)
+  if (distribution_type == "data") {
+    distribution_fields <- setdiff(distribution_fields, c(
+      "expected_values",
+      "x",
+      "dims"
+    ))
+  }
+
+  distribution_input_names <- grep(distribution_name, names(module_input), value = TRUE)
+  for (field in distribution_fields) {
+    set_param_vector(
+      field = field, module = distribution_module,
+      module_input = module_input[distribution_input_names]
+    )
+  }
+
+  switch(distribution_type,
+    "data" = {
+      # Data distribution initialization
+      distribution_module$set_observed_data(linked_ids["data_link"])
+      distribution_module$set_distribution_links(distribution_type, linked_ids["fleet_link"])
+    },
+    "process" = {
+      # Process distribution initialization
+      distribution_module$set_distribution_links("random_effects", linked_ids)
+    },
+    check_valid_input(input = distribution_type, valid_options = valid_distribution_types, arg_name = "distribution_type")
+  )
+
+  # Final message to confirm success
+  cli::cli_inform(c(
+    "i" = "{distribution_name} initialized successfully for {names(distribution_name)}."
+  ))
```

Comment:
Do we want to have success messages for all things eventually? Where, users could turn off this level of information the same way they can turn it off in the creation of FIMSFit? I am hesitant to just have a single example of one though and not have them throughout. Perhaps we can just create a #TODO: statement at the top of this file and the top of create_default_parameters.R that identifies that these should be created in the future for all functions within the files? What do you think @Bai-Li-NOAA?

Reply:
Creating a #TODO: task at the top of the file sounds good to me.

Reply:
Done